StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/MandateTextProvider.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  MandateTextProvider.swift
+//  StripePaymentSheet
+//
+//  Created by Nick Porter on 10/3/24.
+//
+
+import Foundation
+@_spi(STP) import StripeCore
+@_spi(STP) import StripeUICore
+
+protocol MandateTextProvider {
```

Comment:
Why the protocol?  Can we delete this and rename FormMandateProvider to `MandateTextProvider`?

Reply:
Ah okay the test was hidden by default, I see how we use it.  IMO I'd rather have the snapshot tests simply build the actual Provider and use it directly; that way those tests can also exercise the code and they're more realistic.  I'm not sure what the benefit of the protocol is; it's not like this is making network calls that we don't want to deal with.  But I don't feel strongly and am not asking you to re-write all that code!

Reply:
Yep just for testing. I like this kind of pattern! Some of our types (`STPElementsSession` and `STPPaymentMethod`) are quite cumbersome to instantiate just how you want to test a specific scenario. I like that I can focus on testing the embedded view with specific scenarios, then write tests for the provider in `FormMandateProviderTests`.

Reply:
I'd rather solve this:
> Some of our types (STPElementsSession and STPPaymentMethod) are quite cumbersome to instantiate just how you want to test a specific scenario.

by making it easier to instantiate vs. creating the protocol.  

We will need to create STPElementsSession and the IntentConfiguration to initialize EmbeddedPaymentElement anyways - it will need those things - so I don't think it's even buying us anything in the long term by mocking out MandateTextProvider.

Reply:
I think thinking about it more, I feel we aren't going to like having the mock in the long term. It will be strange to initialize EmbeddedPaymentElement w/ STPElementSession + IntentConfiguration _and_ a MandateProvider which uses those same things?

Reply:
Talked offline - actually we probably won't initialize EPE with that stuff!